django/http/request.py

```diff
@@ -366,7 +381,13 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
         if not encoding:
             encoding = settings.DEFAULT_CHARSET
         self.encoding = encoding
+        query_string = query_string or ''
+        parse_qsl_kwargs = {
+            'keep_blank_values': True,
+            'fields_limit': settings.DATA_UPLOAD_MAX_NUMBER_FIELDS,
+        }
         if six.PY3:
+            parse_qsl_kwargs['encoding'] = encoding
```

Comment:
Make the py2 variant support and drop encoding, no need to special case this here